server/plugin.go

```diff
@@ -31,61 +31,67 @@ type Plugin struct {
 }
 
 func (p *Plugin) OnActivate() error {
-	// Ensure we have an enterprise license or a development environment
+	config := p.getConfiguration()
+	if !config.Enabled {
+		p.API.LogInfo("Content moderation is disabled")
+		return nil
+	}
+
 	if !pluginapi.IsEnterpriseLicensedOrDevelopment(
 		p.API.GetConfig(),
 		p.API.GetLicense(),
 	) {
 		return fmt.Errorf("this plugin requires an Enterprise license")
 	}
-	// Setup direct store access
+
 	client := pluginapi.NewClient(p.API, p.Driver)
 	SQLStore, err := sqlstore.New(client.Store, &client.Log)
 	if err != nil {
-		p.API.LogError("cannot create SQLStore", "err", err)
+		p.API.LogError("Cannot create SQLStore", "err", err)
 		return err
 	}
 	p.sqlStore = SQLStore
 
-	return p.initialize()
+	if err := p.initialize(config); err != nil {
+		p.API.LogError("Cannot initialize plugin", "err", err)
+		return nil
+	}
+
+	return nil
 }
 
-func (p *Plugin) initialize() error {
+func (p *Plugin) initialize(config *configuration) error {
 	if p.processor != nil {
 		p.processor.stop()
 		p.processor = nil
 	}
 
-	config := p.getConfiguration()
 	if !config.Enabled {
 		p.API.LogInfo("Content moderation is disabled")
 		return nil
 	}
 
-	botID, err := p.API.EnsureBotUser(&model.Bot{Username: config.BotUsername})
+	moderator, err := initModerator(p.API, config)
 	if err != nil {
-		return errors.Wrap(err, "could not initialize bot user")
+		return errors.Wrap(err, "failed to initialize moderator")
 	}
 
 	excludedUsers := config.ExcludedUserSet()
 	excludedChannels := config.ExcludedChannelSet()
 
 	thresholdValue, err := config.ThresholdValue()
 	if err != nil {
-		p.API.LogError("failed to load moderation threshold", "err", err)
 		return errors.Wrap(err, "failed to load moderation threshold")
 	}
 
-	moderator, err := initModerator(p.API, config)
+	botID, err := p.API.EnsureBotUser(&model.Bot{Username: config.BotUsername})
```

Comment:
This would suggest that every time I change the Bot display name in config I get a new bot.

Reply:
That is the current behavior, yes. I'm not cleaning up old bots - the thinking is that a user could care about their previous history with an old bot. Should we remove them?

Reply:
@wiggin77 I was wrong about this. `EnsureBotUser` stores the bot ID in the plugin's key value store, and only creates one. When I call this with a different username, it just updates the existing bot's username. I don't think this behavior should be fine. WDYT?